### PR TITLE
[DO NOT MERGE] make org-msg work with mu4e

### PR DESCRIPTION
**Pull request by JulienMasson**
Hi Jeremy,

I've been able to use your module with the e-mail client mu4e.
However I had to to rewrite some of your functions because they are using gnus functions.
As my title say, I don't want to merge this patch, it's only to show you how I did to make mu4e work with your module.

I want to discuss with you how we can include my changes "properly".
I have couple ideas in mind:
- hook mechanism
- multiple backends: org-msg.el (core), org-msg-gnus.el, org-msg-mu4e.el ...
- check mail-user-agent var

Btw this module is awesome :)

Signed-off-by: Julien Masson <massonju.eseo@gmail.com>

### CODE REVIEW

File: org-msg.el [hunk 1]
```diff
@@ -276,31 +276,14 @@ Example:
...
-  (cl-flet ((attr-str (x)
-	      (concat " " (symbol-name (car x)) "=\""
-		      (xml-escape-string (cdr x)) "\"")))
-    (if attr
-	(apply 'concat (mapcar #'attr-str attr))
-      "")))
+  (with-current-buffer mu4e~view-buffer-name
+    (let ((body-html (mu4e-message-field-at-point :body-html))
+	  (tmp-file (concat "/tmp/" (mu4e-message-field-at-point :message-id))))
+      (with-current-buffer (find-file-noselect tmp-file)
```
Comment (jeremy-compostella) [inline org-msg.el]: You should use the following construction instead:
(with-temp-buffer
   (insert body-html)
   (write-file tmp-file))

File: org-msg.el [hunk 2]
```diff
@@ -276,31 +276,14 @@ Example:
...
-file."
-  (let* ((pages '())
-	 (save-page (lambda (url &optional _args) (push url pages)))
-	 (browse-url-browser-function save-page))
-    (cl-letf (((symbol-function 'gnus-summary-show-article) #'ignore))
-      (save-window-excursion
-	(gnus-article-browse-html-article)))
-    (substring (car (last pages)) (length "file://"))))
-
-(defun org-msg-attrs-str (attr)
```
Comment (jeremy-compostella) [inline org-msg.el]: Why do you remove this function ?

**Comment by jeremy-compostella** [comment]
This is actually a great news. I realized that many people are are now using **mu4e** mail agent recently.  I looked into it and even if it looks like a great mail agent, I don't find anything that would improve my workflow so I stick to gnus.

As a consequence, I am not capable of doing the port to **mu4e**.  I am really glad that you are doing it as I would be happy to see more people using this module.
I believe that the changes are going to be contained and small based on this patch.  

Now regarding how we could integrate it, I would really appreciate if we can keep everything into one file.  That should not be an issue considering the size of the require changes.

I wanted to experiment so I created the mu4e branch. I have gave you the push permission (I think, you'll tell me if it works).  Be careful as I think it also gave you the right to push to the master branch...
I think this is important we test it well on the mu4e branch to make sure it works well for both of us before we merge it to the master branch as it will automatically update the MELPA version...

I based my approach on the mail-user-agent variable as I believe that it should be good enough for most use cases.  If down the road it does not work for some other use cases, we would just have to rewrite the org-msg-mua-call function.  Feel free to suggest any change, ..., improvement or provide review.

I gave you authorship of the mu4e backend implementation of course.

Whatever the solution we agree on and before we merge to master, there is work to do in the documentation (in org-msg.el and README.org).

**Note:** I'll be traveling next week.

File: org-msg.el (hunk 2 above)
Comment (JulienMasson) [reply org-msg.el]: yes sorry that's a mistake

File: org-msg.el (hunk 1 above)
Comment (JulienMasson) [reply org-msg.el]: got it :)

**Comment by JulienMasson** [comment]
Okay thanks jeremy I'm gonna do that in this dedicated branch.
I'll let you know when it's stable with the backend mu4e :)

**Pull request closed without merging**